docs/configuration.rst

```diff
@@ -79,6 +79,131 @@ Rally will ask you a few more things in the advanced setup:
 * whether or not Rally should keep the Elasticsearch benchmark candidate installation including all data by default. This will use lots of disk space so you should wipe ``~/.rally/benchmarks/races`` regularly.
 
+Configuration File Reference
+----------------------------
+
+Rally stores its configuration in the file ``~/.rally/rally.ini``. It contains of the following sections.
+
+meta
+~~~~
+
+This section contains meta information about the configuration file.
+
+* ``config.version``: The version of the configuration file format. This property is managed by Rally and should not be changed.
+
+system
+~~~~~~
+
+This section contains global information for the current benchmark environment. This information should be identical on all machines where Rally is installed.
+
+* ``env.name`` (default: "local"): The name of this benchmark environment. It is used as meta-data in metrics documents if an Elasticsearch metrics store is configured. Only alphanumeric characters are allowed.
+* ``probing.url`` (default: "https://github.com"): This URL is used by Rally to check for a working Internet connection. It's useful to change this to an internal server if all data are hosted inside the corporate network and connections to the outside world are prohibited.
+* ``available.cores`` (default: number of logical CPU cores): Determines the number of available CPU cores. Rally aims to create one asyncio event loop per core and will distribute clients evenly across event loops.
+* ``async.debug`` (default: false): Enables debug mode on Rally's internal `asyncio event loop <https://docs.python.org/3/library/asyncio-eventloop.html#enabling-debug-mode>`_. This setting is mainly intended for troubleshooting.
+* ``passenv`` (default: "PATH"): A comma-separated list of environment variable names that should be passed to the Elasticsearch process.
+
+node
+~~~~
+
+This section contains machine-specific information.
+
+* ``root.dir`` (default: "~/.rally/benchmarks"): Rally uses this directory to store all benchmark-related data. It assumes that it has complete control over this directory and any of its subdirectories.
+* ``src.root.dir`` (default: "~/.rally/benchmarks/src"): The directory where the source code of Elasticsearch or any plugins is checked out. Only relevant for benchmarks from sources.
+
+source
+~~~~~~
+
+This section contains more details about the source tree.
+
+* ``remote.repo.url`` (default: "https://github.com/elastic/elasticsearch.git"): The URL from which to checkout Elasticsearch.
+* ``elasticsearch.src.subdir`` (default: "elasticsearch"): The local path, relative to ``src.root.dir``, of the Elasticsearch source tree.
+* ``cache`` (default: true): Enables Rally's internal source artifact cache. Artifacts are cached based on their git revision.
```

Comment:
We could give one or two examples in parens about what are source artifacts.
e.g.:

> Enables Rally's internal source artifact (i.e. elasticsearch*.tar.gz and optionally repository-*zip files).

and provide a link to https://esrally.readthedocs.io/en/stable/pipelines.html?highlight=pipelines#from-sources

Reply:
addressed in https://github.com/elastic/rally/pull/1137/commits/415c198c5c289f6533ae27e0895e174cd0f4bbd8